src/Compiler/Microsoft.AspNetCore.Razor.Language/test/RazorDiagnosticDescriptorTest.cs

```diff
@@ -74,4 +77,23 @@ public void RazorDiagnosticDescriptor_HashCodesNotEqual()
         // Assert
         Assert.False(result);
     }
+
+    [Fact]
+    public void NoDuplicateDiagnosticIds()
```

Comment:
Thanks for the test. Looks like we have also ComponentDiagnosticFactory and some more: https://github.com/search?q=repo%3Adotnet%2Frazor+DiagnosticPrefix+%3D+%22RZ%22&type=code. Consider adding those to the test as well.